internal/character/clara/talent.go

```diff
@@ -0,0 +1,153 @@
+package clara
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	TalentCounter key.Modifier = "clara-talent-counter"
+	TalentMark    key.Modifier = "clara-talent-mark"    // MAvatar_Klara_00_PassiveATK_Mark for BPSkill_Revenge
+	TalentRes     key.Modifier = "clara-talent-dmg-res" // MAvatar_Klara_00_Passive_DamageReduce
+)
+
+// Under the protection of Svarog, DMG taken by Clara when hit by enemy
+// attacks is reduced by 10%. Svarog will mark enemies who attack Clara with
+// his Mark of Counter and retaliate with a Counter, dealing Physical DMG
+// equal to *% of Clara's ATK.
+//
+// flow:
+//
+// onBeforeBeingAttacked: + TalentMark, + TalentRes
+//
+// onAfterAttack: + follow-up attack
+func init() {
+	modifier.Register(TalentCounter, modifier.Config{
+		Listeners: modifier.Listeners{
+			// add marker modifier on attacker enemy
+			OnBeforeBeingAttacked: func(mod *modifier.Instance, e event.AttackStart) {
+				mod.Engine().AddModifier(e.Attacker, info.Modifier{
+					Name:   TalentMark,
+					Source: mod.Source(),
+				})
+			},
+
+			// remove marker modifier on all enemies when clara dies
+			OnBeforeDying: func(mod *modifier.Instance) {
+				for _, enemyID := range mod.Engine().Enemies() {
+					mod.Engine().RemoveModifier(enemyID, TalentMark)
+				}
+			},
+		},
+	})
+}
+
+func (c *char) initTalent() {
+	c.engine.AddModifier(c.id, info.Modifier{
+		Name:   TalentCounter,
+		Source: c.id,
+	})
+
+	c.engine.AddModifier(c.id, info.Modifier{
+		Name:   TalentRes,
+		Source: c.id,
+		Stats:  info.PropMap{prop.AllDamageReduce: 0.1},
+	})
+}
+
+// MAvatar_Klara_00_PassiveATK_Mark
+//
+// listens to AttackEnd events and check if clara is eligible for counter
+func (c *char) talentActionEndListener(e event.AttackEnd) {
+	attackerID := e.Attacker
+
+	// won't counter non-enemies (dominated ally)
+	if !c.engine.IsEnemy(attackerID) {
+		return
+	}
+
+	if c.canCounter(e) {
+		c.doCounter(attackerID)
+	}
```

Comment:
The counter marker should be added prior to the `canCounter` check. The cases where the marker is added:
1. When clara gets hit
2. When E6 && an ally gets hit. **Note:** this always happens and is done prior to the 50% fixed change